tensorrt_llm/_torch/pyexecutor/model_engine.py

```
@@ -810,6 +814,58 @@ def _general_warmup(self, resource_manager: ResourceManager,
+        """
+        if self.dist is None:
+            return has_batch
+
+        is_ready = has_batch
+
+        # For large warmup sizes, pre-check GPU memory to avoid OOM that
+        # would leave other ranks stuck in collective ops mid-forward.
+        # The OOM catch in _general_warmup only helps the failing rank;
+        # other ranks remain blocked in the forward pass.
+        if is_ready and num_tokens > 256:
+            free_mem, total_mem = torch.cuda.mem_get_info()
```

Ruff (0.15.10)

[warning] Unpacked variable `total_mem` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)